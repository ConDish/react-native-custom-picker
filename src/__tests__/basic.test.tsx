import { shallow } from 'enzyme'
import * as React from 'react'
import { CustomPicker, CustomPickerProps } from '../'

describe('CustomPicker Basic Tests', () => {
  it('should render correctly with initial props', () => {
    const wrapper = shallow<CustomPickerProps>(
      <CustomPicker options={['1', '2', '3']} />
    )
    expect(wrapper).toMatchSnapshot()

    const customPicker = wrapper.instance()
    expect(customPicker.props.defaultValue).toEqual(undefined)
    expect(customPicker.props.value).toEqual(undefined)
    expect(customPicker.props.placeholder).toEqual('Pick an item...')
    expect(customPicker.props.modalAnimationType).toEqual('none')
    expect(customPicker.props.headerTemplate).toEqual(undefined)
    expect(customPicker.props.footerTemplate).toEqual(undefined)
    expect(customPicker.props.onValueChange).toEqual(undefined)
    expect(customPicker.props.onFocus).toEqual(undefined)
    expect(customPicker.props.onBlur).toEqual(undefined)
  })

  it('should able to select item', () => {
    let selectedValue = null
    const wrapper = shallow<CustomPickerProps>(
      <CustomPicker
        options={['1', '2', '3']}
        onValueChange={value => {
          selectedValue = value
        }}
      />
    )

    const customPicker = wrapper.instance() as CustomPicker
    customPicker.selectOption('3', true)
    expect(selectedValue).toEqual('3')
  })

  it('should able to set default value', () => {
    const wrapper = shallow<CustomPickerProps>(
      <CustomPicker options={['1', '2', '3']} defaultValue={'1'} />
    )
    const customPicker = wrapper.instance()
    expect(customPicker.props.defaultValue).toEqual('1')
  })

  it('should able to set initial value and default value', () => {
    let selectedValue = null
    const wrapper = shallow<CustomPickerProps>(
      <CustomPicker
        options={['1', '2', '3']}
        value={'1'}
        defaultValue={'3'}
        onValueChange={value => {
          selectedValue = value
        }}
      />
    )
    const customPicker = wrapper.instance() as CustomPicker

    expect(customPicker.props.defaultValue).toEqual('3')
    expect(customPicker.props.value).toEqual('1')

    customPicker.selectOption('2', true)
    expect(selectedValue).toEqual('2')

    customPicker.clear()
    expect(selectedValue).toEqual('3')
  })
})
