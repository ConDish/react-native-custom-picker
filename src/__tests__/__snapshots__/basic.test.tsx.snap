// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPicker Basic Tests should render correctly with initial props 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
              "margin": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Pick an item...
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <Component
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.4)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "justifyContent": "center",
                "paddingHorizontal": 20,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "maxHeight": 1274,
                },
                undefined,
              ]
            }
          >
            <ScrollView>
              <TouchableOpacity
                activeOpacity={0.2}
                key="0"
                onPress={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderBottomColor": "grey",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "height": 40,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    1
                  </Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                key="1"
                onPress={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderBottomColor": "grey",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "height": 40,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    2
                  </Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                key="2"
                onPress={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderBottomColor": "grey",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "height": 40,
                        "justifyContent": "space-between",
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    3
                  </Text>
                </View>
              </TouchableOpacity>
            </ScrollView>
          </View>
        </View>
      </View>
    </TouchableWithoutFeedback>
  </Component>
</View>
`;
